libwallet/src/slate.rs

```diff
@@ -514,7 +518,7 @@ impl Slate {
 		// Generate a random kernel offset here
 		// and subtract it from the blind_sum so we create
 		// the aggsig context with the "split" key
-		self.tx_or_err_mut()?.offset = match use_test_rng {
+		let my_offset = match use_test_rng {
```

Comment:
can't this be written as an if then else ? or doesn't that work at expression level?

Reply:
Stylistic choice, I tend to prefer match statements over if-else (particularly when the block is returning a value) because I find there's less tendency for them to get confused once logic starts expanding